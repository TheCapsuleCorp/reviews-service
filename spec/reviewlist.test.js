import React from 'react';
import jest from 'jest';
import renderer from 'react-test-renderer';
import ReviewList from '../public/components/reviewList.jsx';


test('ReviewList/ReviewListEntry components render based on the passed in reviews array', () => {

  const reviews = [
    {
      comments: "Great place, ideal location.",
      created_at: "2018-02-01T20:36:24Z",
      first_name: "Tyler",
      has_profile_pic: true,
      identity_verified: true,
      listing_id: 21826970,
      month: "February",
      picture_url: "https://a0.muscache.com/im/pictures/0794b769-abf6-4127-ae9f-3c3bf53fa034.jpg?aki_policy=profile_x_medium",
      year: "2018"
    },
    {
      comments: "The apartment is in a great location. A lot of bars and restaurants around that you can walk to.",
      created_at: "2017-12-02T20:11:08Z",
      first_name: "Jackie",
      has_profile_pic: true,
      identity_verified: false,
      listing_id: 21826970,
      month: "December",
      picture_url: "https://a0.muscache.com/im/pictures/45a0be0d-7f73-4046-9d57-134457f2a311.jpg?aki_policy=profile_x_medium",
      year: "2017"
    }
  ];

  const reviewComponents = renderer
    .create(<ReviewList reviews={reviews} />)
    .toJSON();

  expect(reviewComponents).toMatchSnapshot();
});
