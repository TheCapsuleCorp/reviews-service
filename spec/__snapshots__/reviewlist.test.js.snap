// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList/ReviewListEntry components render based on the passed in reviews array 1`] = `
<div
  className="reviews-section"
>
  <div
    className="reviews-container"
  >
    <div
      className="reviews-head"
    >
      <img
        src="https://a0.muscache.com/im/pictures/0794b769-abf6-4127-ae9f-3c3bf53fa034.jpg?aki_policy=profile_x_medium"
      />
      <div
        className="reviews-head-text"
      >
        <span
          className="reviews-head-name"
        >
           
          Tyler
           
        </span>
        <span
          className="reviews-head-date"
        >
           
          February
           
          2018
           
        </span>
      </div>
    </div>
    <div
      className="reviews-body"
    >
      Great place, ideal location.
    </div>
  </div>
  <div
    className="reviews-container"
  >
    <div
      className="reviews-head"
    >
      <img
        src="https://a0.muscache.com/im/pictures/45a0be0d-7f73-4046-9d57-134457f2a311.jpg?aki_policy=profile_x_medium"
      />
      <div
        className="reviews-head-text"
      >
        <span
          className="reviews-head-name"
        >
           
          Jackie
           
        </span>
        <span
          className="reviews-head-date"
        >
           
          December
           
          2017
           
        </span>
      </div>
    </div>
    <div
      className="reviews-body"
    >
      The apartment is in a great location. A lot of bars and restaurants around that you can walk to.
    </div>
  </div>
  <div
    className="reviews-nav"
  >
    <button
      className="reviews-btn reviews-empty-btn"
    >
      .
    </button>
    <button
      className="reviews-btn reviews-btn-1 reviews-btn-selected"
      onClick={[Function]}
    >
      1
    </button>
  </div>
</div>
`;
